### ISSUE TITLE
View#load_texture doc example contains error

### ISSUE THREAD

**Issue by GordonMcIntyre** [SketchUp/api-issue-tracker#445]
The documentation for View.load_texture here:

http://ruby.sketchup.com/Sketchup/View.html#load_texture-instance_method

contains the following example:

```ruby
module Example
  class MyTool

    def activate
      view = Sketchup.active_model.active_view
      image_rep = view.model.materials.current.texture.image_rep
      @texture_id = view.load_texture(image_rep)
    end

    def deactivate(view)
      view.release_texture(@texture_id)
    end

    def draw(view)
      points = [ [0, 0, 0], [9, 0, 0], [9, 9, 0], [0, 9, 0] ]
      uvs = [ [0, 0, 0], [1, 0, 0], [1, 1, 0], [0, 1, 0] ]
      view.draw(points, texture_id: @texture_id, uvs: uvs)
    end

  end
end
Sketchup.active_model.select_tool(Example::MyTool.new)
```
In the draw method, the call to view.draw omits the first parameter for openglenum. I also think the hash key for :texture_id should be :texture and I'm going to log that as a separate issue.

So this line:
```ruby
view.draw(points, texture_id: @texture_id, uvs: uvs)
```
Should be something like:
```ruby
view.draw(GL_QUADS, points, texture: @texture_id, uvs: uvs)        
```

**Comment by DanRathbun** [comment]
**Please note** that the same erroneous code example is given for method ...

[`Sketchup::View#release_texture`](https://ruby.sketchup.com/Sketchup/View.html#release_texture-instance_method)

**Comment by DanRathbun** [comment]
Okay, after testing the limited example in the docs ([`Sketchup::View#load_texture`](https://ruby.sketchup.com/Sketchup/View.html#load_texture-instance_method) and [`Sketchup::View#release_texture`](https://ruby.sketchup.com/Sketchup/View.html#release_texture-instance_method)) is **not a full working example** !

Here is an example that works (and uses better practice) and fixes the issue #446 (wrong option key for texture id) …

```ruby
module Example
  class MyTool

    def activate
      view = Sketchup.active_model.active_view
      @points = [ [0, 0, 0], [9, 0, 0], [9, 9, 0], [0, 9, 0] ]
      @uvs = [ [0, 0, 0], [1, 0, 0], [1, 1, 0], [0, 1, 0] ]
      @extents = Sketchup.active_model.bounds.add(@points)
      @color = Sketchup::Color.new(0, 0, 255)
      matl = view.model.materials.current
      # The current material might be nil just after model load.
      # Also the current selected material may not have a texture:
      if matl && matl.texture
        image_rep = matl.texture.image_rep
        @texture_id = view.load_texture(image_rep)
      else
        @texture_id = nil
      end
    rescue => err
      puts err.inspect
    ensure
      view.invalidate # redraw the view immediately
    end

    def deactivate(view)
      view.release_texture(@texture_id) if @texture_id
      view.invalidate # redraw the view when done
    end

    def draw(view)
      view.line_stipple = '' # Solid line
      view.drawing_color = @color
      if @texture_id
        view.draw(GL_QUADS, @points, texture: @texture_id, uvs: @uvs)
      else
        view.draw(GL_QUADS, @points)
      end
    end

    def getExtents
      @extents
    end

    def onMouseMove(flags, x, y, view)
      view.invalidate # redraw the view on mousemove
    end

  end
end

Sketchup.active_model.select_tool(Example::MyTool.new)
```

**Comment by DanRathbun** [comment]
Okay then ... I've just posted the **working** and **safer** example to the public forum ...
https://forums.sketchup.com/t/view-draw-texture-working-example/148519
